### PR TITLE
[MKC-1469] GIGA R1 WiFI Dual Core Debugging

### PR DESCRIPTION
**Pull request by Hannes7eicher**
## What This PR Changes
This PR adds an article explaining how to debug both the M7 and the M4 core on the GIGA R1 WiFI simultaneously with a J-Link debugger and Ozone.

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+Your J-link should come with a cable included and if you look closer you will see that one side is marked with red, indicating the orientation of the cable. The red side of the cable should be on the same side as the VCC pin on the SWD pad. To know the exact orientation of your cable be sure to check the respective documentation or datasheet of your debugger but you can start with the orientation as shown below and if it doesn't work you can try placing it differently until you get a working connection. Once we start using the debugger software in the following steps you will know if the cable is placed correctly.
+
+## Software 
+
+### Setting up with the Arduino IDE
+
+First, if you haven't done it yet, install the [Arduino IDE](https://www.arduino.cc/en/software) and connect your GIGA R1 WiFi. When uploading a sketch to your Arduino board with the Arduino IDE, it will build an `.ELF` file of the sketch. We will need this file to debug in Ozone in the next steps. To easily find the file path of the `.ELF` file, we can enable the show verbose output option in the Arduino IDE. To do this, open up the preferences under **File > Preferences** in the Arduino IDE.
+
```
Comment (pedromsousalima) [inline content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md:52]: If you have just installed the core you need to reset the IDE to make sure all options are visible to you. This includes the ability to upload to a second core.

File: content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+## Connection
+
+Connecting the J-link to your GIGA R1 WiFi is super easy because there are special pins for debugging labeled as **SWD** pins.
+
+![SWD Pins](./assets/jtag_pins.png)
+
+Your J-link should come with a cable included and if you look closer you will see that one side is marked with red, indicating the orientation of the cable. The red side of the cable should be on the same side as the VCC pin on the SWD pad. To know the exact orientation of your cable be sure to check the respective documentation or datasheet of your debugger but you can start with the orientation as shown below and if it doesn't work you can try placing it differently until you get a working connection. Once we start using the debugger software in the following steps you will know if the cable is placed correctly.
+
```
Comment (pedromsousalima) [inline content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md:46]: Missing image

File: content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+Inside the first instance:
+
+- Select **Flash split** > **1MB M7 + 1MB M4**.
+
+- Select **Main Core** as **Target core**.
+
+- Upload **"Sketch M7"** (see below).
+
+Inside the seconds instance:
```
Comment (pedromsousalima) [inline content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md:81]:
```suggestion
Inside the second instance:
```

**Review by pedromsousalima — CHANGES_REQUESTED** [review]
Looks good! Only small changes needed  :)

File: content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md (hunk 2 above)
Comment (Hannes7eicher) [reply content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md:48]: I added a link to Segger's documentation instead.

**Pull request merged**